Dart/testData/generate/EqualsAndHashCode1.txt

```diff
@@ -5,18 +5,18 @@ class EqualsAndHashCode1 {
   static var s;
 
   @override
-  bool operator ==(Object other) {
-    if (identical(this, other)) {
-      return true;
-    }
-    return other is EqualsAndHashCode1 &&
-        this.str == other.str &&
-        this.v == other.v &&
-        this.i == other.i;
-  }
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+          other is EqualsAndHashCode1 &&
+              str == other.str &&
+              v == other.v &&
+              i == other.i;
 
   @override
-  int get hashCode {
-    return str.hashCode ^ v.hashCode ^ i.hashCode;
-  }<caret>
+  int get hashCode =>
+      super.hashCode ^
```

Comment:
Thanks!
I think we can go without any options for now.
Two checks `other is ThisClass &&  runtimeType == other.runtimeType` seem to be redundant. It's enough to have the first one if we allow subclasses, and the second if we don't. Agree?
My preference is for `runtimeType == other.runtimeType` as it gives better guarantees to meet the transitivity requirement imposed by the equals contract.
`super == other` required in any case.